.github/workflows/python-test-package.yml

```diff
@@ -0,0 +1,58 @@
+# This workflow will install Python dependencies, run tests and lint with a variety of Python versions
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: Python package using marketplace setup-miniconda
+
+on: [push, workflow_dispatch]
```

Comment:
I think we need `pull_request` too? So that the pull requests are tested as well.

Reply:
You could add a status check to see branch is up to date before allowing merge. This would mean each PR is tested through the push workflow